lua/default-config.lua

```diff
@@ -159,7 +166,7 @@ require("core.zen").config()
 require("core.telescope").config()
 require("core.treesitter").config()
 require("core.which-key").config()
-require("core.nvimtree").config()
+configure_plugin "core.nvimtree"
```

Comment:
I would just keep the function call in `nvimtree.config()`. I don't see why we can't keep there tbh.

Reply:
So I understand you correct, the statement require("core.nvimtree") returns null before the nvimtree plugin is downloaded in :PackerInstall. Hence, we cannot call config() on a null object. As I understand we need to check if the plugin exist prior to calling config (if the nvimtree.config() is called from default-config.lua), the alternative would be to call config() from  nvimtree.lua at line 5, after the call check.

It would be great if you can elaborate a bit more about this comment.

Reply:
My idea is to change `configure_plugin` to be `is_plugin_loaded` and then call it from inside the usual `M.config()` so it would maybe look something like this: 

```lua
M.config = function()
  if lv-utils.is_plugin_loaded('nvimtree') == false then
    return
  end
  O.plugin.nvimtree = {
   side = "left",
   ---
   --
   }
end    
```

I think we probably also need to move the first `pcall` into the setup function, which I think is probably the real culprit here.
https://github.com/ChristianChiarulli/LunarVim/blob/11ac140eb57c70dd30b5ec34d31da1679c4151ce/lua/core/nvimtree.lua#L2

Reply:
Cool, thanks, I'll give that a try.

Reply:
I tried to update nvimtree.lua according to the comment above, but if I leave the 
require("core.nvimtree").config() in default-config.lua then I stilll get the Error at the initial PackagerInstall (before the plugins are loaded).

The config in nvimtree is used for defining the configurable parameters, so I also successfully tried to move calling config to the plugin as.
  -- NvimTree
  use {
    "kyazdani42/nvim-tree.lua",
    -- event = "BufWinOpen",
    -- cmd = "NvimTreeToggle",
    commit = "fd7f60e242205ea9efc9649101c81a07d5f458bb",
    config = function()
      require("core.nvimtree").config()
      require("core.nvimtree").setup()
    end,
  }

Therefore the two options that works for me are.

#1. Move the nvimtree  require("core.nvimtree").config() from default-config.lua / default-config and call the require("core.nvimtree").config()  from the plugin.lua

#2. Keep the config() call in default-config.lua / default-config and call lv_utils.configure_plugin()
to use pcall before config() is called.


Comments welcome.